src/consensus.rs

```diff
@@ -184,6 +185,10 @@ impl Consensus {
                 )
                 .context("Failed to add peer")?;
         }
+        // Only first peer has itself as a voter in the initial conf state.
+        // This needs to be propagated manually to other peers as it is not contained in any log entry.
+        state_ref.set_first_voter(all_peers.first_peer_id);
+        state_ref.set_conf_state(ConfState::from((vec![all_peers.first_peer_id], vec![])))?;
```

Comment:
Does it mean, that we force the first peer in a cluster to be a voter no mater what?
Won't it affect the cluster once we implement peer removing?

Reply:
It just provides the initial conf state. When peer starts the sync - either through log entries or through snapshot it will get the current conf state, even if it doesn't have the first peer.

The problem was that the changes could not be applied to the empty conf state, the first transitions already assumed that conf state had one exact voter in it.

Reply:
So currently I don't think that it will present a problem